parlai/core/image_featurizers.py

```diff
@@ -251,4 +256,229 @@ def load(self, path):
+            return torch.load(new_path)
+
+
+class DetectronFeatureExtractor:
+    """
+    Code adapted from https://github.com/facebookresearch/mmf/blob/master/tools/scripts/
+    features/extract_features_vmb.py.
+
+    Docstrings and type annotations added post hoc.
+    """
+
+    MAX_SIZE = 1333
+    MIN_SIZE = 800
+
+    def __init__(self, opt: Opt, use_cuda: bool = False):
+        self.opt = opt
+        self.use_cuda = use_cuda
+        self.num_features = 100
+
+        try:
+            import cv2
+
+            self.cv2 = cv2
+        except ImportError:
+            raise ImportError("Please install opencv: pip install opencv-python")
+        try:
+            import maskrcnn_benchmark  # noqa
+        except ImportError:
+            raise ImportError(
+                'Please install vqa-maskrcnn-benchmark to use faster_r_cnn_152_32x8d features: '
+                '1. git clone https://gitlab.com/vedanuj/vqa-maskrcnn-benchmark.git\n'
+                '2. cd vqa-maskrcnn-benchmark\n'
+                '3. python setup.py develop'
+            )
+        self._build_detection_model()
+
+    def _build_detection_model(self):
+        """
+        Build the detection model.
+
+        Builds a CNN using the vqa-maskrcnn-benchmark repository.
+        """
+        from maskrcnn_benchmark.config import cfg
+        from maskrcnn_benchmark.modeling.detector import build_detection_model
+        from maskrcnn_benchmark.utils.model_serialization import load_state_dict
+
+        dp = self.opt['datapath']
+        build(dp)
+        cfg_path = os.path.join(dp, 'models/detectron/detectron_config.yaml')
+        model_path = os.path.join(dp, 'models/detectron/detectron_model.pth')
+
+        cfg.merge_from_file(cfg_path)
+        cfg.freeze()
+
+        model = build_detection_model(cfg)
+        checkpoint = torch.load(model_path, map_location=torch.device("cpu"))
+
+        load_state_dict(model, checkpoint.pop("model"))
+
+        if self.use_cuda:
+            model.to("cuda")
+        model.eval()
+        self.detection_model = model
+
+    def _image_transform(
+        self, img: "Image"
+    ) -> Tuple[torch.Tensor, float, Dict[str, int]]:
+        """
+        Using Open-CV, perform image transform on a raw image.
+
+        :param img:
+            raw image to transform
+
+        :return (img, scale, info):
+            img: tensor representation of image
+            scale: scale of image WRT self.MIN_SIZE & self.MAX_SIZE
+            info: dict containing values for img width & height
+        """
+        im = np.array(img).astype(np.float32)
+
+        if im.shape[-1] > 3:
+            im = np.array(img.convert("RGB")).astype(np.float32)
+
+        # IndexError: too many indices for array, grayscale images
+        if len(im.shape) < 3:
+            im = np.repeat(im[:, :, np.newaxis], 3, axis=2)
+
+        im = im[:, :, ::-1]
+        im -= np.array([102.9801, 115.9465, 122.7717])
```

Comment:
Hmm - is it stated in the MMF code what this is? Some constant to center the distribution probably?

Reply:
I didn't see any documentation around it, perhaps @apsdehal or @vedanuj could offer some insight